### PR TITLE
#500 Added auto-generated keptn cli docs

### PR DESCRIPTION
**Pull request by christian-kreuzberger-dtx**
This is a first draft of auto-generated cli docs for the keptn cli.

The sourcecode for generating the docs will be within the keptn cli itself, e.g. `keptn generate docs`

I also updated all links that previously went to `reference/cli/` to `cli/commands/`.

Preview: https://deploy-preview-501--keptn.netlify.com/docs/develop/cli/commands/

### CODE REVIEW

File: content/docs/develop/cli/commands/_index.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+title: Commands
+description: Auto-generated documentation of Keptn CLI
```
Comment (johannes-b) [inline content/docs/develop/cli/commands/_index.md:3]: When talking about the `keptn cli`, let`s use a lower case `k`.

File: content/docs/develop/cli/_index.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
+---
+title: keptn CLI
+description: Explains how to download and install the keptn CLI as well as which commands are available.
+weight: 30
+icon: help
+keywords: [cli, setup]
+---
+
+In this section, the functionality and commands of the Keptn CLI are described. The Keptn CLI allows installing, configuring, and
```
Comment (johannes-b) [inline content/docs/develop/cli/_index.md:9]: In this paragraph: 

When talking about the `keptn cli`, let`s use a lower case `k`.

File: content/docs/develop/cli/_index.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+uninstalling Keptn. Furthermore, the CLI allows creating projects, onboarding services, and sending new artifact events.
+
+## Prerequisites
+- All prerequisites from the [setup](../../installation/setup-keptn#prerequisites) are needed.
+
+## Install the Keptn CLI
+Every release of Keptn provides binaries for the Keptn CLI. These binaries are available for Linux, macOS, and Windows.
```
Comment (johannes-b) [inline content/docs/develop/cli/_index.md:16]: When talking about the `keptn cli`, let`s use a lower case `k`.

File: content/docs/develop/cli/_index.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+1. Find the `keptn` binary in the unpacked directory.
+  - Linux / macOS
+    
+    add executable permissions (``chmod +x keptn``), and move it to the desired destination (e.g. `mv keptn /usr/local/bin/keptn`)
+
+  - Windows
+
+    move/copy the executable to the desired folder and, optionally, add the executable to the PATH environment variable for a more convenient experience.
+
+4. Now, the Keptn CLI help can be shown by:
```
Comment (johannes-b) [inline content/docs/develop/cli/_index.md:29]: When talking about the `keptn cli`, let`s use a lower case `k`.

File: content/docs/develop/cli/_index.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    move/copy the executable to the desired folder and, optionally, add the executable to the PATH environment variable for a more convenient experience.
+
+4. Now, the Keptn CLI help can be shown by: 
+    ```console
+    keptn --help
+    ```
+
+## Start using the Keptn CLI
+
+In the following, the commands provided by the Keptn CLI are described.
```
Comment (johannes-b) [inline content/docs/develop/cli/_index.md:36]: When talking about the `keptn cli`, let`s use a lower case `k`.

File: content/docs/develop/cli/_index.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+  - Windows
+
+    move/copy the executable to the desired folder and, optionally, add the executable to the PATH environment variable for a more convenient experience.
+
+4. Now, the Keptn CLI help can be shown by: 
+    ```console
+    keptn --help
+    ```
+
+## Start using the Keptn CLI
```
Comment (johannes-b) [inline content/docs/develop/cli/_index.md:34]: When talking about the `keptn cli`, let`s use a lower case `k`.

**Comment by christian-kreuzberger-dtx** [comment]
@johannes-b I have updated my PR.

Please note that we have many many many occurences of `Keptn CLI` in our existing docs and within the actual keptn CLI golang code. I have not introduced those with my PR, they have been there before, I copied the docs where possible.

I have opened two other issues for this to get this fixed in the future:

* https://github.com/keptn/keptn/issues/1718
* https://github.com/keptn/keptn.github.io/issues/525

**Pull request merged**